sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -36,14 +38,12 @@ class LongHashSet extends org.apache.spark.util.collection.OpenHashSet[Long]
  * expressions.
  */
 abstract class CodeGenerator[InType <: AnyRef, OutType <: AnyRef] extends Logging {
-  import scala.reflect.runtime.{universe => ru}
   import scala.reflect.runtime.universe._
 
-  import scala.tools.reflect.ToolBox
-
-  protected val toolBox = runtimeMirror(getClass.getClassLoader).mkToolBox()
+  protected val cbe = CompilerFactoryFactory.getDefaultCompilerFactory().newClassBodyEvaluator()
 
   protected val rowType = typeOf[Row]
+  protected val exprType = typeOf[Expression]
   protected val mutableRowType = typeOf[MutableRow]
   protected val genericRowType = typeOf[GenericRow]
   protected val genericMutableRowType = typeOf[GenericMutableRow]
```

Comment:
`SpecificMutableRow` is generated for a given schema, the layout will be optimized (by compiler) for primitive types.

We can re-visit it later, this PR just change the code template from Scala AST to Java code.